include/layers/ReshapeLayer.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <vector>
+
+#include "layers/Layer.hpp"
+#include "layers/Tensor.hpp"
+
+namespace it_lab_ai {
+
+class ReshapeLayer : public Layer {
+ public:
+  explicit ReshapeLayer(bool allowzero = false,
```

Comment:
ONNX `allowzero`: when 1, zeros in target shape copy input dims. Ensure this semantics and validate product with `-1`; add tests.